applications/schema/mutations.py

```diff
@@ -42,6 +43,108 @@
 from .types import BerthApplicationNode, WinterStorageApplicationNode
 
 
+def _validate_boat_fields_common(application_data):
+    if "boat_id" in application_data:
+        if any(
+            field.startswith("boat_") and field != "boat_id"
+            for field in application_data
+        ):
+            raise VenepaikkaGraphQLError(
+                _(
+                    'Cannot use both "boatId" and other "boatXXX" field(s) at the same time."'
```

Comment:
Small nitpick: for consistency with the rest of our error messages, could you rename all this `boatXXX` to snake case `boat_xxx`? 🙏